doc/aaa/TACACS+ Design.md

```diff
@@ -0,0 +1,544 @@
+
+ - SONiC currently supported TACACS+ features:
+   - Authentication.
+   - User session authorization.
+   - User session accounting.
+   - User command authorization with local permission.
+
+ - New features:
+   - User command authorization with TACACS+ server.
+   - User command accounting with TACACS+ server.
+
+# 1 Functional Requirement
+## 1.1 User command authorization
+ - Authorization when user run any executable file or script on SONiC host.
+   - The full path and parameters will be send to TACACS+ server side for authorization.
+   - For recursive command/script, only the top level command have authorization.
+   - No authorization for  bash built-in command and bash function, but if a bash function call any executable file or script, those executable file or script will have authorization.
+
+ - TACACS+ authorization is configurable:
+   - TACACS+ authorization can be enable/disable.
+   - TACACS+ authorization method will communicate with TACACS+ server for authorization, TACACS+ server should setup permit/deny rules.
+
+ - Failover:
+   - If a TACACS+ server not accessible, the next TACACS+ server authorization will be performed.
+   - When all remote TACACS+ server not accessible, TACACS+ authorization will failed.
+   - When TACACS+ authorization failed, fallover is configurable:
+     - Disable local authorization as failover, then user can't run any command.
+     - Enable local authorization as failover, then user can run command with local authorization.
+     - For local authorization, please check [TACACS+ Authentication](#TACPLUS-Authentication).
+
+## 1.2 User command accounting
+ - Accounting is the action of recording what a user is doing, and/or has done.
+
+ - Following event will be accounted:
+   - Command start event.
+   - Command finish event.
+
+ - User command in Docker will not be accounted.
+   - User command in docker actually run by docker service, so we can't identify if command are run by user or system service.
+
+ - Support TACACS+ accounting and local accounting:
+   - TACACS+ will send event to TACACS+ server, and communication will be encrypted, for more detail please check [RFC8907](#rfc8907).
+   - Local accounting will save event to syslog.
+   - Both TACACS+ accounting and local accounting are configurable.
+
+ - User secrets not exist in accounting event:
+   - Use regex in /etc/sudoers PASSWD_CMDS to identify user secrets.
+   - User secret will be replaced with ***.
+
+## 1.3 User script
+ - User can create and run their own script. 
+ - If user create a script but TACACS+ service side not have configuration to allow user run this script, user script will be blocked by authorization.
+
+## 1.4 Docker support
+ - Docker exec command will be covered by Authorization and Accounting. 
+ - Any command run inside docker container will not covered by Authorization and Accounting.
```

Comment:
Should move/add this to the limitations section. 
A user can run commands inside a container to evade TACACS authorization and accounting. 

It may be good to add a recommendation to administrators to disable access to 'docker exec' for most users.

Reply:
Fixed.